functions/packages.bash

```diff
@@ -141,473 +180,467 @@ homegear_setup() {
+  if {
     rm -rf /etc/nginx/sites-enabled/default
     cp "$BASEDIR"/includes/nginx.conf /etc/nginx/sites-enabled/openhab
-
     sed -i "s/DOMAINNAME/${domain}/g" /etc/nginx/sites-enabled/openhab
-
-    if [ "$AUTH" = true ]; then
-      echo "Installing password utilities..."
-      apt-get -y -q install apache2-utils || FAILED=true
-      echo "Creating password file..."
-      htpasswd -b -c /etc/nginx/.htpasswd "$username" "$password"
-      uncomment "#AUTH" /etc/nginx/sites-enabled/openhab
-    fi
-
-    if [ "$SECURE" = true ]; then
-      if [ "$VALIDDOMAIN" = true ]; then
-        certbotpackage="python3-certbot-nginx"
-        if is_debian || is_raspbian; then
-          gpg --keyserver pgpkeys.mit.edu --recv-key 8B48AD6246925553
-          gpg -a --export 8B48AD6246925553 | apt-key add -
-          gpg --keyserver pgpkeys.mit.edu --recv-key 7638D0442B90D010
-          gpg -a --export 7638D0442B90D010 | apt-key add -
-          if is_jessie; then
-            certbotrepo="jessie-backports"
-            certbotpackage="certbot"
-          elif is_stretch; then
-            certbotrepo="stretch-backports"
-          fi
-          certbotoption="-t"
-          if [ -n "$certbotrepo" ]; then
-            echo -e "# This file was added by openHABian to install certbot\\ndeb http://ftp.debian.org/debian ${certbotrepo} main" > /etc/apt/sources.list.d/backports.list
-            apt-get update
-          fi
-        elif is_ubuntu; then
-          apt -get -y -q --force-yes install software-properties-common
-          add-apt-repository ppa:certbot/certbot
-        fi
-        echo "Installing certbot..."
-        apt-get -y -q --force-yes install "${certbotpackage}" "${certbotoption}" "${certbotrepo}"
-        mkdir -p /var/www/$domain
-        uncomment "#WEBROOT" /etc/nginx/sites-enabled/openhab
-        nginx -t && service nginx reload
-        echo "Creating Let's Encrypt certificate..."
-        certbot certonly --webroot -w /var/www/$domain -d $domain || FAILED=true #This will cause a prompt
-        if [ "$FAILED" = false ]; then
-          certpath="/etc/letsencrypt/live/$domain/fullchain.pem"
-          keypath="/etc/letsencrypt/live/$domain/privkey.pem"
+  }; then echo "OK"; else echo "FAILED"; fi
+
+  if [ "$auth" == "true" ]; then
+    cond_echo "Setting up Nginx password options..."
+    echo -n "$(timestamp) [openHABian] Installing Nginx password utilities..."
+    if cond_redirect apt-get install --yes install apache2-utils; then echo "OK"; else echo "FAILED"; return 1; fi
+    if cond_redirect htpasswd -b -c /etc/nginx/.htpasswd "$nginxUsername" "$nginxPass"; then echo "OK"; else echo "FAILED (password file)"; return 1; fi
+    uncomment "#AUTH" /etc/nginx/sites-enabled/openhab
+  fi
+
+  if [ "$secure" == "true" ]; then
+    cond_echo "Setting up Nginx security options..."
+    if [ "$validDomain" == "true" ]; then
+      echo -n "$(timestamp) [openHABian] Installing certbot..."
+      if is_ubuntu; then
```

Comment:
Is this PR supposed to remove Ubuntu specific code remainders ?
Or is it already about (new) Ubuntu support ?

Reply:
That was merely a piece that was already in place and I figured we will need to support it at some point so there was not need to remove it.

Reply:
yeah see 2 comments down on Ubuntu. I think you can leave it inside.